### PR TITLE
screenpipe 0.1.98 (new cask)

### PR DESCRIPTION
**Pull request by Neptune650**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

This PR adds a cask for the Screenpipe app.

### CODE REVIEW

**Comment by chenrui333** [comment]
- https://github.com/Homebrew/homebrew-core/pull/196345

**Comment by Neptune650** [comment]
> * [screenpipe 0.1.98(new formula)  homebrew-core#196345](https://github.com/Homebrew/homebrew-core/pull/196345)

@chenrui333 Hello, firstly that formula doesn't build from source - and the maintainer of the project agreed to using a cask instead.

File: Casks/s/screenpipe.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  sha256 arm:   "2cb4d60689ee22bf7c639e75cbcb57d3b8f21732fc383ff57592b4044b5e4396",
+         intel: "969dad8fdbf79af2fa6ea14b604ac656041d8fa0281777b70bac6893d13bd34b"
+
+  url "https://github.com/mediar-ai/screenpipe/releases/download/v#{version}/screenpipe-#{version}-#{arch}-apple-darwin.tar.gz"
+  name "screenpipe"
+  desc "Library to build personalized AI powered by what you've seen, said, or heard"
+  homepage "https://github.com/mediar-ai/screenpipe"
+
+  auto_updates true
+  depends_on formula: "ffmpeg"
```
Comment (louis030195) [inline Casks/s/screenpipe.rb:14]: bun?

Comment (louis030195) [reply Casks/s/screenpipe.rb:14]: oh

**Comment by daeho-ro** [comment]
There is a desktop app, but cannot find a way to download without payment.

**Comment by Neptune650** [comment]
> There is a desktop app, but cannot find a way to download without payment.

@daeho-ro The GitHub releases provide macOS downloads for free.

**Comment by daeho-ro** [comment]
@Neptune650 Then serve the binary through the formula and desktop app by cask, that looks better. How do you think?

**Comment by Neptune650** [comment]
> @Neptune650 Then serve the binary through the formula and desktop app by cask, that looks better. How do you think?

Well after talking to the developer we agreed that distribution screenpipe entirely through a cask would be more convenient overall, the screenpipe building process is pretty complicated as is.

File: Casks/s/screenpipe.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  container type: :naked
```
Comment (bevanjkay) [inline Casks/s/screenpipe.rb:15]:
```suggestion
```

This should not be required.

File: Casks/s/screenpipe.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  zap trash: "~/.screenpipe"
+
+  def caveats
+    <<~EOS
+      Bun is not available on Homebrew
+      To install Bun, please visit the official website:
+      https://bun.sh
+      Follow the instructions there to install Bun on your system.
+    EOS
+  end
```
Comment (bevanjkay) [inline Casks/s/screenpipe.rb:30]: I'm hesitant to include this in `homebrew-cask` if it requires additional dependencies that we can't fulfill at install time. Any other thoughts @Homebrew/cask?

**Comment by SMillerDev** [comment]
> screenpipe author here, the goal is to make CLI available through brew install screenpipe (skipping the brew tap), this is unrelated to the desktop app

App is both open-source and CLI-only (i.e. it only uses the binary artifact). In that case, and in the spirit of deduplication, submit it first to homebrew/core as a formula that builds from source. If it is rejected, you may then try again as a cask (link to the issue from your pull request so we can see the discussion and reasoning for rejection).

https://docs.brew.sh/Acceptable-Casks#rejected-casks

File: Casks/s/screenpipe.rb (hunk 3 above)
Comment (p-linnane) [reply Casks/s/screenpipe.rb:30]: No, we shouldn't ship a cask like this.

**Comment by p-linnane** [comment]
@SMillerDev is correct that this needs to be added to Homebrew/core instead of Homebrew/cask.

**Pull request closed without merging**